projects/igniteui-angular/src/lib/drop-down/drop-down.component.spec.ts

```diff
@@ -1582,316 +908,360 @@ describe('IgxDropDown ', () => {
+                spyOn(dropdown.onClosed, 'emit').and.callThrough();
+                tabs.triggerEventHandler('click', UIInteractions.clickEvent);
+                tick();
+                fixture.detectChanges();
+                expect(dropdown.onOpening.emit).toHaveBeenCalledTimes(1);
+                expect(dropdown.onOpened.emit).toHaveBeenCalledTimes(1);
+                let dropdownItems = fixture.debugElement.queryAll(By.css(`.${CSS_CLASS_ITEM}`));
+                dropdownItems[2].triggerEventHandler('click', UIInteractions.clickEvent);
+                tick();
+                fixture.detectChanges();
+                expect(dropdown.onClosing.emit).toHaveBeenCalledTimes(1);
+                expect(dropdown.onClosed.emit).toHaveBeenCalledTimes(1);
 
-    public toggleDropDown() {
-        this.dropdownDisabledAny.toggle();
-    }
+                input.triggerEventHandler('click', UIInteractions.clickEvent);
+                tick();
+                fixture.detectChanges();
+                expect(dropdown.onOpening.emit).toHaveBeenCalledTimes(2);
+                expect(dropdown.onOpened.emit).toHaveBeenCalledTimes(2);
+                dropdownItems = fixture.debugElement.queryAll(By.css(`.${CSS_CLASS_ITEM}`));
+                dropdownItems[1].triggerEventHandler('click', UIInteractions.clickEvent);
+                tick();
+                fixture.detectChanges();
+                expect(dropdown.onClosing.emit).toHaveBeenCalledTimes(2);
+                expect(dropdown.onClosed.emit).toHaveBeenCalledTimes(2);
 
-    public selectItem5() {
-        this.dropdownDisabledAny.setSelectedItem(4);
-    }
-}
+                img.triggerEventHandler('click', UIInteractions.clickEvent);
+                tick();
+                fixture.detectChanges();
+                expect(dropdown.onOpening.emit).toHaveBeenCalledTimes(3);
+                expect(dropdown.onOpened.emit).toHaveBeenCalledTimes(3);
+                dropdownItems = fixture.debugElement.queryAll(By.css(`.${CSS_CLASS_ITEM}`));
+                dropdownItems[0].triggerEventHandler('click', UIInteractions.clickEvent);
+                tick();
+                fixture.detectChanges();
+                expect(dropdown.onClosing.emit).toHaveBeenCalledTimes(3);
+                expect(dropdown.onClosed.emit).toHaveBeenCalledTimes(3);
+            }));
+        });
+    });
+});
 
 @Component({
     template: `
-    <button (click)="toggleDropDown()">Show</button>
-    <button (click)="selectItem5()">Select 5</button>
-    <igx-drop-down #dropdownDisabled>
-        <igx-drop-down-item [igxDropDownItemNavigation]="dropdownDisabled" *ngFor="let item of items"
+    <button (click)="toggleDropDown()">Toggle</button>
+    <igx-drop-down id="test-id" igxDropDownItemNavigation [maxHeight]="maxHeight"
+    [displayDensity]="density" (onSelection)="onSelection($event)" [allowItemsFocus]="true"
+    (onOpening)="onToggleOpening($event)" (onOpened)="onToggleOpened()"
```

Comment:
This are not needed as the tests using directly the drop down events:
onToggleOpening, onToggleOpened, onSelection, onToggleClosing, onToggleClosed
You can remove them from the definition.